### DISCUSSION TITLE
"28*28 masks",  what is that?

### DISCUSSION THREAD

**Discussion by xuexl** [facebookresearch/detectron2#3393] [Q&A] [answered]
When i deploy the model in libtorch, I get a pred_mask that is a 28*28`s tensor.
pred_masks: (1,1,.,.) =
 Columns 1 to 6  6.7040e-02  8.0095e-02  8.0535e-02  8.8935e-02  1.4882e-01  2.1055e-01
  6.6355e-02  8.8262e-02  8.7877e-02  1.0452e-01  1.7061e-01  2.4665e-01
  4.9940e-02  7.5910e-02  1.0294e-01  1.1402e-01  2.1387e-01  2.9677e-01
  5.3084e-02  9.5872e-02  1.1876e-01  1.3956e-01  2.3670e-01  3.2493e-01
  6.6633e-02  1.2497e-01  1.7715e-01  1.9293e-01  3.6666e-01  4.7469e-01
  8.7384e-02  1.9130e-01  2.4560e-01  2.8409e-01  4.4529e-01  5.5942e-01
  1.0172e-01  2.3206e-01  3.2975e-01  4.0185e-01  6.1649e-01  6.8876e-01
...

What is that ?  How to use it ? 
I find some information in the site "https://detectron2.readthedocs.io/en/latest/tutorials/deployment.html"
But, I think it is so not clear and so hard to understand.

**Comment by tsubauaaa** [comment]
I also want to know. 
I want to display the mask image from the mask of float [] obtained by libtorch.
Please tell me how to do it. How do I convert this float [] to binary mask?

**Reply by xuexl** [reply]
You need resize the mask of 28*28 to the size of the ROI. Like this,  
 
` cv::Mat cv_mask(28,28,CV_32FC1);
   std::memcpy(cv_mask.data, pred_masks[i].data<float>(), 28*28*sizeof(float));
   cv::resize(cv_mask, cv_mask, cv::Size(x2-x1,y2-y1));`

**Comment by ppwwyyxx** [comment, marked as answer]
It is the mask of the object inside the given bounding box, i.e. left of this image:
<img width="938" alt="paste" src="https://user-images.githubusercontent.com/1381301/138785683-cd95116a-d77c-473f-bec8-0c6a67d68dc9.png">


detectron2 provides this function https://detectron2.readthedocs.io/en/latest/modules/layers.html#detectron2.layers.paste_masks_in_image that performs this

**Comment by tsubauaaa** [comment]
Thanks to everyone, I was able to create a mask image.
<img src="https://user-images.githubusercontent.com/11596443/139408445-8d907470-9ef9-4e1a-8768-4477789252eb.png" width="300">

**Reply by Jain-Archit** [reply]
Hi, 
Can you share the code with which you were able to get the mask image from 28x28 sized mask.
